test/octree/test_octree.cpp

```diff
@@ -962,6 +962,414 @@ TEST (PCL, Octree_Pointcloud_Iterator_Test)
   // check if every branch node and every leaf node has been visited
   ASSERT_EQ (octreeA.getLeafCount (), leafNodeCount);
   ASSERT_EQ (octreeA.getBranchCount (), branchNodeCount);
+
+
+  // breadth-first at a given depth iterator test
+
+  lastDepth = 0;
+  branchNodeCount = 0;
+  leafNodeCount = 0;
+
+  leafNodeVisited = false;
+
+  OctreePointCloud<PointXYZ>::FixedDepthIterator bfdIt;
+  const OctreePointCloud<PointXYZ>::FixedDepthIterator bfdIt_end = octreeA.fixed_depth_end ();
+
+  for (unsigned int depth = 0; depth <= octreeA.getTreeDepth (); ++depth)
+  {
+    for (bfdIt = octreeA.fixed_depth_begin (depth); bfdIt != bfIt_end; ++bfdIt)
+    {
+      unsigned int current_depth = bfdIt.getCurrentOctreeDepth ();
+      ASSERT_TRUE (lastDepth <= current_depth);
+      ASSERT_TRUE (depth == current_depth);
+      lastDepth = current_depth;
+
+      if (bfdIt.isBranchNode ())
+      {
+        branchNodeCount++;
+        ASSERT_FALSE (leafNodeVisited);
+      }
+
+      if (bfdIt.isLeafNode ())
+      {
+        leafNodeCount++;
+        leafNodeVisited = true;
+      }
+    }
+  }
+
+  // check if every branch node and every leaf node has been visited
+  ASSERT_EQ (octreeA.getLeafCount (), leafNodeCount);
+  ASSERT_EQ (octreeA.getBranchCount (), branchNodeCount);
+
+  // Test the operator==() and operator!=()
+  OctreePointCloud<PointXYZ>::FixedDepthIterator itA_end = octreeA.fixed_depth_end ();
+  OctreePointCloud<PointXYZ>::FixedDepthIterator itB_end = octreeA.fixed_depth_end ();
+
+  ASSERT_TRUE (itA_end == itB_end);
+  ASSERT_FALSE (itA_end != itB_end);
+  ASSERT_TRUE ((itA_end == itB_end) == !(itA_end != itB_end));
+
+
+  // Test the operator==() and operator!=() from different octree
+  OctreePointCloud<PointXYZ> octreeB (1.0f); // low resolution
+
+  // add point data to octree
+  octreeB.setInputCloud (cloudIn);
+
+  itA_end = octreeA.fixed_depth_end ();
+  itB_end = octreeB.fixed_depth_end ();
+
+  // ASSERT_FALSE (itA_end == itB_end);
+  // ASSERT_TRUE (itA_end != itB_end);
+  // ASSERT_TRUE ((itA_end == itB_end) == !(itA_end != itB_end));
+}
+
+TEST (FixedDepthIterator, Constructor)
+{
+  // create octree instance
+  OctreeBase<int> octree;
+
+  // set octree depth
+  octree.setTreeDepth (1);
+
+  // generate some voxel indices
+  for (unsigned char childIndex = 0; childIndex < 8; ++childIndex)
+  {
+    OctreeKey key;
+    key.pushBranch (childIndex);
+
+    octree.createLeaf (key.x, key.y, key.z);
+  }
+
+  OctreeBase<int>::FixedDepthIterator it = octree.fixed_depth_begin();
+
+  const OctreeKey key = it.getCurrentOctreeKey ();
+  ASSERT_EQ(key.x, 0);
```

Comment:
Spacing in this and the other asserts below.

Reply:
Deleted